src/Interfaces/QuoteProgressValidator.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Jauntin\SavingQuote\Interfaces;
+
+interface QuoteProgressValidator
+{
+    /**
+     * @param array<string, string|array|mixed> $data
+     * @return array<int, array|mixed>
+     */
+    public function rules(array $data): array;
+
+    /**
+     * @param array<string, string|array|mixed> $data
+     * @return array<int, array|mixed>|array{}
+     */
+    public function validate(array $data): array;
```

Comment:
Same here. What does this return? Does it throw a ValidationException like a Laravel validator, and generally behave like one? 

Or if it isn't like a Laravel validator, then the developer needs documentation to know what this return array is so they know what to do.
Its entire shape, like if it's an array of errors, then is the array like `['field' => 'fieldName', 'errors' => ['tooSmall' => 'value is too small', 'tooBig' => 'value is too big'], /* other properties? ... */]`?

Reply:
Honestly, here's my guess at a reasonable implementation. Instead of making our own thing, we can just do something like: the app provides a set of rules, and we validate those rules. Something like this:

```php

// Rules provides an array of Laravel validation rules
interface QuoteProgressValidationRules { public function rules(): array; }

class QuoteProgressController {
    public function single(string $hash, QuoteProgressService $service, ?QuoteProgressValidationRules $rules = null)
    {
        /// ...do other things ...

        if ($rules) {
            Validator::make($data, $rules)->validate();
        }

        // ... return the data...
    }
}
```

Reply:
Also, once again, do we really want to support not having rules? Wouldn't it be better if this integration failed if we forgot to make a rules class?

Assuming I'm correct, then the rules will always be provided, and we don't need it to be nullable and we don't need to check if it exists.